frontend/appflowy_flutter/lib/plugins/document/presentation/favorite/favorite_button.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy/workspace/application/favorite/favorite_bloc.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra_ui/style_widget/hover.dart';
+import 'package:flowy_infra_ui/widget/flowy_tooltip.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class DocumentFavoriteButton extends StatelessWidget {
+  const DocumentFavoriteButton({
+    super.key,
+    required this.view,
+  });
+
+  final ViewPB view;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<FavoriteBloc, FavoriteState>(
+      builder: (context, state) {
+        final isFavorite = state.views.contains(view);
+        return _buildFavoriteButton(context, isFavorite);
+      },
```

Comment:
`ViewPB` is a generated class I don't equality check would work on objects, might be the reason icon does not get updated, better approach would be to do this

`final isFavorite = state.views.map((v)=> v.id)).toList().contains(view.id)`

Reply:
Yess it worked!, thanks a lot